### PR TITLE
Add the Promoted add-ons shelf to the home page

### LINKED ISSUE #9518: Add the Promoted add-ons shelf to the home page
Depends on https://github.com/mozilla/addons-server/issues/14923

The component for the new Promoted Add-ons shelf will land in #9517, but it doesn't appear anywhere. This issue is about adding the shelf to the home page.

From the PRD:
- Impressions and clicks of add-ons shown in this section must be recorded in Google Analytics.

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #9518 

Screen Shot 2020-09-11 at 10 14 06 AM

### CODE REVIEW

File: src/amo/pages/Home/index.js [hunk 1]
```diff
@@ -235,6 +240,14 @@ export class HomeBase extends React.Component {
 
     const showHeroPromo = clientApp !== CLIENT_APP_ANDROID;
 
+    const { promotedExtensions } = shelves;
+    if (Array.isArray(promotedExtensions)) {
+      // If there are fewer than 6 promoted extensions, just use the first 3.
```
Comment (eviljeff) [inline src/amo/pages/Home/index.js:245]: Interesting... is this a requirement?  If someone's paying for promotion I would have thought we'd always want to show as many as possible.

File: src/amo/sagas/home.js [hunk 2]
```diff
@@ -122,6 +124,16 @@ export function* fetchHomeData({
       },
     };
 
+    const promotedExtensionsParams: SearchParams = {
+      api: state.api,
+      filters: {
+        addonType: ADDON_TYPE_EXTENSION,
+        page_size: String(LANDING_PAGE_PROMOTED_EXTENSION_COUNT),
+        promoted: SPONSORED,
+        sort: SEARCH_SORT_RANDOM,
```
Comment (eviljeff) [inline src/amo/sagas/home.js:133]: no `app` filter?  Or is that added elsewhere?

**Review by eviljeff — APPROVED** [review]

File: src/amo/pages/Home/index.js (hunk 1 above)
Comment (bobsilverberg) [reply src/amo/pages/Home/index.js:245]: This is from the PRD. I guess they don't want the shelf to look unbalanced with 4 or 5 extensions.

File: src/amo/sagas/home.js (hunk 2 above)
Comment (bobsilverberg) [reply src/amo/sagas/home.js:133]: Oh yeah. This shelf is for desktop only, so I guess I should pass `app=firefox`. Thanks.

Comment (bobsilverberg) [reply src/amo/sagas/home.js:133]: Actually, that *is* added elsewhere, so we don't need it here.

**Pull request merged**